pkg/redirectpolicy/manager.go

```diff
@@ -250,7 +250,11 @@ func (rpm *Manager) OnUpdatePodLocked(pod *slimcorev1.Pod) {
 	if err != nil {
 		return
 	}
-	podData := rpm.getPodMetadata(pod, podIPs)
+	var podData *podMetadata
+	if podData, err = rpm.getPodMetadata(pod, podIPs); err != nil {
+		log.Debugf("failed to get valid pod metadata %s", err)
+		return
```

Comment:
>Debug messages will never be printed in prod environments, so if this error condition is ever hit then the user has no way to confirm why the LRP doesn't work.

The checks done by `getPodMetadata` (and similar existing code in cilium [1] ) more of back-up? These checks are already done by Kubernetes. But I'm fine with changing the log level for the error cases to error instead of debug.

```
kubectl apply -f examples/kubernetes-local-redirect/backend-pod.yaml
The Pod "lrp-pod" is invalid: spec.containers[0].ports[1].protocol: Unsupported value: "uddp": supported values: "SCTP", "TCP", "UDP"
```

[1] https://github.com/cilium/cilium/blob/ada53a91a98618d24dad2abda97dd9c60923003c/pkg/k8s/watchers/pod.go#L637

Reply:
Looking at `updatePodHostData()` and `getPodMetadata()`, the checks look almost identical. Should `updatePodHostData()` be guaranteed to be called before hitting `getPodMetadata()`? Do you know of any way this would not be the case?

Overall, if we validate the inputs earlier on and prevent hitting these cases, then there's no point in doing additional validation in `getPodMetadata()` and this whole error checking question becomes moot.

Reply:
The checks might be required for the code path where we fetched pods from the `podStore`, in which case `updatePodHostData` isn't called.

Reply:
OK, makes sense. The changes here LGTM.